cli/internal/typer/orchestrator.go

```diff
@@ -0,0 +1,150 @@
+package typer
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/rudderlabs/rudder-iac/cli/internal/app"
+	"github.com/rudderlabs/rudder-iac/cli/internal/logger"
+	"github.com/rudderlabs/rudder-iac/cli/internal/typer/generator/core"
+	"github.com/rudderlabs/rudder-iac/cli/internal/typer/generator/platforms/kotlin"
+	"github.com/rudderlabs/rudder-iac/cli/internal/typer/plan"
+)
+
+var (
+	orchestratorLog = logger.New("typer.orchestrator")
+)
+
+// Orchestrator coordinates the RudderTyper code generation process
+type Orchestrator struct {
+	deps app.Deps
+}
+
+// GenerationOptions contains configuration for code generation
+type GenerationOptions struct {
+	TrackingPlanID string
+	Platform       string
+	OutputPath     string
+}
+
+// NewOrchestrator creates a new RudderTyper orchestrator
+func NewOrchestrator(deps app.Deps) *Orchestrator {
+	return &Orchestrator{
+		deps: deps,
+	}
+}
+
+// Generate orchestrates the complete code generation process
+func (o *Orchestrator) Generate(ctx context.Context, options GenerationOptions) error {
+	orchestratorLog.Debug("starting code generation",
+		"trackingPlanID", options.TrackingPlanID,
+		"platform", options.Platform,
+		"outputPath", options.OutputPath)
+
+	// Step 1: Fetch tracking plan data
+	fmt.Println("(mock) 📥 Fetching tracking plan data...")
+	trackingPlan, err := o.fetchTrackingPlan(ctx, options.TrackingPlanID)
+	if err != nil {
+		return fmt.Errorf("fetching tracking plan: %w", err)
+	}
+	orchestratorLog.Debug("(mock) fetched tracking plan", "rules", len(trackingPlan.Rules))
```

Comment:
With a PlanProvider interface in place, this doesn't have to be mock, it can be a direct call to the PlanProvider's function